martin/src/utils/id_resolver.rs

```diff
@@ -57,63 +60,72 @@ impl IdResolver {
         } else {
             Some(unique_name.clone())
         };
-        let new_name = self.resolve_int(name, unique_name);
-        if name != new_name {
-            warn!(
-                "Source `{name}`{info} was renamed to `{new_name}`. Source IDs must be unique, cannot be reserved, and must contain alpha-numeric characters or `._-`",
-                info = info.map_or(String::new(), |v| format!(" ({v})"))
-            );
-        }
-        new_name
-    }
-
-    #[must_use]
-    fn resolve_int(&self, name: &str, unique_name: String) -> String {
-        // Ensure name has no prohibited characters like spaces, commas, slashes, or non-unicode etc.
-        // Underscores, dashes, and dots are OK. All other characters will be replaced with dashes.
-        let mut name = name.replace(
-            |c: char| !c.is_ascii_alphanumeric() && c != '_' && c != '.' && c != '-',
-            "-",
-        );
+        let stanitised_name = Self::sanitise(name);
```

Comment:
I am not super happy with the current code. This is still messy.

I don't know what would be better as the code is very performance optimised which makes it hard to be "pretty" as well.

Reply:
this code is only used once, no need to make it super performant